sig-and-wg/sig-core/proposals/introducing_knative_brokers.md

```diff
@@ -0,0 +1,90 @@
+
+1. Allow Kyma users to try the new Knative `Broker` based eventing.
+2. Keep current users of Event Bus CRDs unaffected and allow a smoother migration path.
+3. Ease our migration and upgrade pain by giving users a Kyma release where they can migrate their lambdas themselves and understand how it behaves.
+
+### Solution 1: Knative Brokers Only
+
+#### Overview
+
+The usage of Knative Eventing shifts the eventing paradigm in Kyma, from a single messaging Pub/Sub system, to an eventing mesh with isolated fault domains, access control and dynamic event routing where many senders inject events into the mesh from multiple source points, and many sinks (i.e subscribers) receive all or a subset of those events based on filters and access control. The responsiblity of Kyma system is to correctly and dynamically wire the eventing mesh to match the user eventing needs and achieve optimum event routing.
+
+Solution 1 describes how Kyma system will build and wire the eventing mesh to achieve proper event routing from external application to the subscribers such as the lambda functions.
+
+#### Architecture
+
+For every namespace that is allowed to receive events, there is a default `Broker` in the namespace. Kyma will do all the wiring from the `Event Service` to the local namespace `Brokers` and subscribers who want to receive events, should create a Knative `Trigger` against their local default `Broker`.
+
+![architecture](assets/knative_brokers_arch.svg)
+
+1. For every new `Application` the `Application Operator` creates an `Event Service` and its corresponding  `Knative Channel` that will be used to fanout events to all allowed brokers.
+2. For every new namespace, Kyma Console ensures it has the label `knative-eventing-injection=enabled` so that a default `Broker` gets created by the Knative Eventing controller.
+3. Once the `Application Broker` gets a provision request for a `Service Instance` which has `bindable` parameter set to `false`, it will create a `Knative Subscription` that has:
+   - The channel pointing to the relevant `Event Service` fanout `Channel`
+   - The subscriber pointing to the default `Broker` of the namespace
+   - The ownerRef set as the `Service Instance` that just got provisioned which ensures the subscription gets deleted by Kubernetes garbage collector when the `Service Instance` gets deprovisioned.
+4. `Kyma Console` creates the proper `Trigger` objects for the `Lambda` function for every event type chosen as a lambda trigger.
+5. Events received by `Event Service` will be delivered to its channel which in turn will do a fanout to all connected brokers which will finally deliverd it to the lambda functions.
+
```

Comment:
I think it would make sense to add the deprovisioning process in a different section. We need to answer:
- Who deletes the Knative Subscription to the fanout channel? IMO the `application` should be the owner of the Knative Channel.

In the Knative Brokers approach, it is not clear how are we solving this:
> - As a developer, I want to publish events to Kyma from inside the cluster.